karpenter-operator/controllers/aws/ec2_nodeclass.go

```diff
@@ -0,0 +1,376 @@
+package aws
```

Comment:
let's move this within the existing karpenter/ folder and let it be nodeclass_controller. Or put it under nodeclass/ folder

Reply:
moved under nodeclass/ and renamed to `ec2_nodeclass_controller`, so we can differentiate it from azure `aks_nodeclass_controller` when we add that.